src/jit/lsra.cpp

```diff
@@ -6709,6 +6809,13 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                     }
                     inVarToRegMap[assignedInterval->getVarIndex(compiler)] = REG_STK;
                 }
+#ifdef _TARGET_ARM_
+                // Consider overlapping floating point register for TYP_DOUBLE
+                else if (!assignedInterval->isConstant && isSecondHalfReg(physRegRecord, assignedInterval))
+                {
+                    assert(assignedInterval->assignedReg->regNum == REG_PREV(physRegRecord->regNum));
```

Comment:
Thanks! I was also having hard time with this condition.
Your suggestion is much clear and make sense :)